data-service/src/main/java/uk/gov/gchq/palisade/service/data/service/SimpleDataService.java

```diff
@@ -128,8 +131,17 @@ DataReaderRequest constructReaderRequest(final ReadRequest request) {
 
                 LOGGER.debug("Output stream closed, {} processed and {} returned, auditing success with audit service", recordsProcessed.get(), recordsReturned.get());
                 auditRequestComplete(readerRequest, recordsProcessed, recordsReturned);
-            } catch (IOException | NoCapacityException ex) {
+            } catch (IOException | RuntimeException ex) {
+                LOGGER.error("Exception thrown while reading data, auditing exception", ex);
                 auditRequestReceivedException(dataRequest, ex);
+                throw new ReadException(ex);
+            } catch (Error err) {
+                // Either an auditRequestComplete or auditRequestException MUST be called here, so catch a broader set of Exception classes than might be expected
+                // Generally this is a bad idea, but we need guarantees of the audit - ie. malicious attempt at StackOverflowError
+                LOGGER.error("Error thrown while reading data, attempting auditing error", err);
```

Comment:
Again don't need to log and throw error. Suggest just audit logging here and then throw error to be acted on elsewhere. Or just change the logging line to log the exception message rather than the full exception